src/Functions/tupleElement.cpp

```diff
@@ -121,58 +126,53 @@ class FunctionTupleElement : public IFunction
             array_offsets.push_back(array_col->getOffsetsPtr());
         }
 
-
-        const DataTypeTuple * input_type_as_tuple = checkAndGetDataType<DataTypeTuple>(input_type);
-        const ColumnTuple * input_col_as_tuple = checkAndGetColumn<ColumnTuple>(input_col);
-
-        if (input_type_as_tuple && input_col_as_tuple)
+        ColumnPtr res;
+        if (const DataTypeTuple * input_type_as_tuple = checkAndGetDataType<DataTypeTuple>(input_type))
         {
+            const ColumnTuple & input_col_as_tuple = checkAndGetColumn<ColumnTuple>(*input_col);
             std::optional<size_t> index = getTupleElementIndex(arguments[1].column, *input_type_as_tuple, arguments.size());
 
             if (!index.has_value())
                 return arguments[2].column;
 
-            ColumnPtr res = input_col_as_tuple->getColumnPtr(index.value());
-
-            /// Wrap into Arrays
-            for (auto it = array_offsets.rbegin(); it != array_offsets.rend(); ++it)
-                res = ColumnArray::create(res, *it);
-
-            if (input_arg_is_const)
-                res = ColumnConst::create(res, input_rows_count);
-
-            return res;
+            res = input_col_as_tuple.getColumnPtr(index.value());
         }
-
-
-        const DataTypeQBit * input_type_as_qbit = checkAndGetDataType<DataTypeQBit>(input_type);
-        const ColumnQBit * input_col_as_qbit = checkAndGetColumn<ColumnQBit>(input_col);
-
-        if (input_type_as_qbit && input_col_as_qbit)
+        else if (const DataTypeQBit * input_type_as_qbit = checkAndGetDataType<DataTypeQBit>(input_type))
         {
+            const ColumnQBit & input_col_as_qbit = checkAndGetColumn<ColumnQBit>(*input_col);
             std::optional<size_t> index = getQBitElementIndex(arguments[1].column, *input_type_as_qbit, arguments.size());
 
             if (!index.has_value())
                 return arguments[2].column;
 
-            ColumnPtr res = assert_cast<const ColumnTuple &>(input_col_as_qbit->getTupleColumn()).getColumnPtr(index.value());
-
-            /// Wrap into Arrays
-            for (auto it = array_offsets.rbegin(); it != array_offsets.rend(); ++it)
-                res = ColumnArray::create(res, *it);
-
-            if (input_arg_is_const)
-                res = ColumnConst::create(res, input_rows_count);
+            res = assert_cast<const ColumnTuple &>(input_col_as_qbit.getTupleColumn()).getColumnPtr(index.value());
+        }
+        else if (const DataTypeObject * input_type_as_object = checkAndGetDataType<DataTypeObject>(input_type))
+        {
+            const auto * subcolumn_name_col = checkAndGetColumnConst<ColumnString>(arguments[1].column.get());
+            if (!subcolumn_name_col)
+                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Second argument of {} with {} first argument must be a constant String", getName(), input_type->getName());
```

Comment:
Since we check this in `getReturnTypeImpl`, do we need to check here again?

Reply:
Just in case it'a always better to check here again. We could have a bug somewhere else that can lead to wrong arguments in `executeImpl`.